### PR TITLE
NPA-2676: Receive QuestionnaireResponse

### PR DESCRIPTION
**Pull request by chris-young-12-nhs**
## Summary
* :sparkles: New Feature - first cut POST QuestionnaireResponse endpoint

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/validated-relationships-service-api.yaml [hunk 1]
```diff
@@ -199,6 +230,236 @@ paths:
                 $ref: '#/components/schemas/OperationOutcome'
```
Comment (JackPlowman) [inline specification/validated-relationships-service-api.yaml:230]: Comment not related to this line

We have expression defined for the operational outcome but none of our errors currently return it, so maybe it should be removed
Operational Outcome confluence page -> https://nhsd-confluence.digital.nhs.uk/pages/viewpage.action?pageId=791874870
Operational Outcome object -> https://github.com/NHSDigital/validated-relationships-service-api/pull/78/files#diff-e143931886bb4215c74e499a54357c5e498c103ff719c91bc8e7a977b3768517R827

File: specification/validated-relationships-service-api.yaml [hunk 2]
```diff
@@ -199,6 +230,236 @@ paths:
...
+          # On Confluence the example has this as a string but here:
+          # https://simplifier.net/packages/hl7.fhir.r4.core/4.0.1/files/81730/~details
+          # it looks like an object
+        source: 
+          type: object
+          description: The person who answered the questions about the subject.
+          properties:
+            type:
+              type: string
+              enum: [RelatedPerson] # Here: https://simplifier.net/packages/hl7.fhir.r4.core/4.0.1/files/81189/~details this looks like it should be a URI
```
Comment (JackPlowman) [inline specification/validated-relationships-service-api.yaml:257]: Don't understand what does this doing, is this correct?

File: specification/validated-relationships-service-api.yaml [hunk 3]
```diff
@@ -199,6 +230,236 @@ paths:
...
+          description: The FHIR resource type
+          enum: ["QuestionnaireResponse"]
+        status:
+          type: string
+          description: The position of the questionnaire response within its overall lifecycle.
+          enum: ["completed"]
+        authored: 
+          type: string
+          description: The date and/or time that this set of answers were last changed.
+          # On Confluence the example has this as a string but here:
```
Comment (JackPlowman) [inline specification/validated-relationships-service-api.yaml:248]: Note: all comments can't be seen on Swagger editor so I presume also developer portal. Close this comment that's the expected behaviour

File: specification/validated-relationships-service-api.yaml [hunk 4]
```diff
@@ -149,6 +149,37 @@ servers:
   - url: 'https://api.service.nhs.uk/validated-relationships/FHIR/R4'
     description: Production environment.
 paths:
+  /QuestionnaireResponse:
```
Comment (JackPlowman) [inline specification/validated-relationships-service-api.yaml:152]: I presume this is ok being above `/RelatedPerson`. Should the list be in alphabetical order?

File: specification/validated-relationships-service-api.yaml [hunk 5]
```diff
@@ -149,6 +149,37 @@ servers:
...
+    post:
+      summary: New Access request
+      description: |
+        ## Overview
+        For any new access request, the necessary details should be collected from a user facing service e.g. 
+        Proxy Access Service and submitted as a QuestionaireResponse. 
+        
+        For the most part demographics information doesn't need to be provided in the access request since it can be pulled from PDS.
```
Comment (JackPlowman) [inline specification/validated-relationships-service-api.yaml:160]: Does this need a section about the sandbox try it now like `/RelatedPerson` or should it be removed from `/RelatedPerson` and the comment moved to a common area

**Review by JackPlowman — CHANGES_REQUESTED** [review]
Looks good, a couple of questions/formatting suggestions

File: specification/validated-relationships-service-api.yaml [hunk 6]
```diff
@@ -199,6 +230,236 @@ paths:
                 $ref: '#/components/schemas/OperationOutcome'
 components:
   schemas:
+    QuestionnaireResponse:
+      type: object
+      description: A FHIR QuestionnaireResponse
+      properties:
```
[the remark this answers is not in the document]
Comment (chris-young-12-nhs) [reply specification/validated-relationships-service-api.yaml:236]: Ah yes, I was wondering if there was a way of declairing these for all paths. Will have a look.

File: specification/validated-relationships-service-api.yaml (hunk 3 above)
Comment (chris-young-12-nhs) [reply specification/validated-relationships-service-api.yaml:248]: Yes just for internal consumption

File: specification/validated-relationships-service-api.yaml (hunk 2 above)
Comment (chris-young-12-nhs) [reply specification/validated-relationships-service-api.yaml:257]: It's not clear to me either. In the example on confluence it is just `RelatedPerson` as you say but the spec seems to be expecting a URI to a specific `RelatedPerson`

File: specification/validated-relationships-service-api.yaml [hunk 7]
```diff
@@ -149,6 +149,50 @@ servers:
...
+          application/fhir+json:
+            schema:
+              $ref: '#/components/schemas/QuestionnaireResponse'
+      responses:
+        '200':
+          description: Request was received successfully for processing
+          content:
+            application:fhir+/json:
+              schema:
+                $ref: '#/components/schemas/OperationOutcome'
```
Comment (JackPlowman) [inline specification/validated-relationships-service-api.yaml:179]: This ref is incorrect as the response shouldn't be an operational outcome

File: specification/validated-relationships-service-api.yaml (hunk 4 above)
Comment (chris-young-12-nhs) [reply specification/validated-relationships-service-api.yaml:152]: I put it above RelatedPerson for alphabetical order. I don't think there is a strict rule.

File: specification/validated-relationships-service-api.yaml (hunk 5 above)
Comment (chris-young-12-nhs) [reply specification/validated-relationships-service-api.yaml:160]: Will add it once it is added to the sandbox

File: specification/validated-relationships-service-api.yaml (hunk 4 above)
Comment (JackPlowman) [reply specification/validated-relationships-service-api.yaml:152]: I should have remembered that R comes after Q 😞

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Pull request merged**